topdown/http_test.go

```diff
@@ -129,22 +130,31 @@ func TestHTTPGetRequestTlsInsecureSkipVerify(t *testing.T) {
 
 	resultObj := ast.MustInterfaceToValue(expectedResult)
 
+	type httpsStruct struct {
+		note     string
+		rules    []string
+		expected interface{}
+		//expectedError error
+	}
+
 	// run the test
-	tests := []struct {
-		note          string
-		rules         []string
-		expected      interface{}
-		expectedError error
-	}{
-		{note: "http.send", rules: []string{fmt.Sprintf(
-			`p = x { http.send({"method": "get", "url": "%s", "force_json_decode": true}, x) }`, ts.URL)}, expected: &Error{Message: fixupDarwinGo118("x509: certificate signed by unknown authority", `x509: “Acme Co” certificate is not trusted`)}},
-		{note: "http.send", rules: []string{fmt.Sprintf(
-			`p = x { http.send({"method": "get", "url": "%s", "force_json_decode": true, "tls_insecure_skip_verify": true}, resp); x := clean_headers(resp) }`, ts.URL)}, expected: resultObj.String()},
-		// This case verifies that `tls_insecure_skip_verify`
-		// is still applied, even if other TLS settings are
-		// present.
-		{note: "http.send", rules: []string{fmt.Sprintf(
-			`p = x { http.send({"method": "get", "url": "%s", "force_json_decode": true, "tls_insecure_skip_verify": true, "tls_use_system_certs": true,}, resp); x := clean_headers(resp) }`, ts.URL)}, expected: resultObj.String()},
+	tests := []httpsStruct{}
+	tests = append(tests, httpsStruct{note: "http.send", rules: []string{fmt.Sprintf(
+		`p = x { http.send({"method": "get", "url": "%s", "force_json_decode": true, "tls_insecure_skip_verify": true}, resp); x := clean_headers(resp) }`, ts.URL)}, expected: resultObj.String()})
+
+	// This case verifies that `tls_insecure_skip_verify`
+	// is still applied, even if other TLS settings are
+	// present.
+	tests = append(tests, httpsStruct{note: "http.send", rules: []string{fmt.Sprintf(
+		`p = x { http.send({"method": "get", "url": "%s", "force_json_decode": true, "tls_insecure_skip_verify": true, "tls_use_system_certs": true,}, resp); x := clean_headers(resp) }`, ts.URL)}, expected: resultObj.String()})
+
+	if runtime.GOOS == "darwin" {
```

Comment:
Why do we need this?

Reply:
code is keep the same with before.
but the "darwin" case special proces.

Reply:
Thanks. Can you please clarify why the special case is needed?

Reply:
[crypto/x509: Verify on macOS](https://github.com/golang/go/issues/57427) and [crypto/x509: Verify on macOS](https://github.com/golang/go/issues/56891) may the changed in 1.19.5.

Reply:
About ```fixupDarwinGo118 , //go:build !go1.18 || !darwin
// +build !go1.18 !darwin```  
You means to add a new method?

I have no solve with it about go version: 1.19.5 in macos https  ..
It returns not a error , but ''.
How did it do well in that situation?

Reply:
Sorry, I'll have another look tomorrow! 👀

Reply:
So something that used to be an error is no longer an error, but only on macos+go1.19.5? That's quite odd. But yeah, let's do something about it -- I'm not too worried because I think few people run OPA as server on macos.

That said, the code here doesn't look right: if `runtime.GOOS != "darwin"`, the fixupDarwinGo118 doesn't make sense in the other branch. 🤔

Reply:
@srenatus 
thanks for spending time to invest it..
So , We skip the macos test?
Delete a test case in 'fixupDarwinGo118'?
Is that ok?

Reply:
Anything that makes the tests pass 😅

Reply:
Just skip the macos test ```fixupDarwinGo118 ```
😅😅